Mpuncel/allocate scheduler

mpuncel opened the pull request:
This gives us a starting point to build our allocation server endpoint

spenceral commented on pkg/grpc/scheduler/protos/scheduler.proto:
```diff
@@ -0,0 +1,17 @@
+package scheduler_protos;
+
+service P2Scheduler {
+  rpc Allocate (AllocateRequest) returns (AllocateResponse) {}
```
nit: I like `AllocateNodes` which I think you mentioned in person earlier

mpuncel commented:
addressed feedback, although I realized that maybe the implementation of the scheduler interface shouldn't be `ApplicatorScheduler` because it's going to be doing grpc stuff not just labels queries? can probably punt that to a different PR once we start the implementation tho
